src/app/base/components/TagSelector/TagSelector.test.tsx

```diff
@@ -65,43 +79,39 @@ describe("TagSelector", () => {
         tags={tags}
       />
     );
-    expect(
-      component.find('[data-testid="selected-tag"] span').at(0).text()
-    ).toBe("tag1");
+    expect(screen.getByTestId("selected-tag")).toHaveTextContent("tag1");
   });
 
   it("opens the dropdown when input is focused", () => {
-    const component = shallow(
+    render(
       <TagSelector
         label="Tags"
         onTagsUpdate={jest.fn()}
         placeholder="Select or create tags"
         tags={tags}
       />
     );
-    expect(component.find(".tag-selector__dropdown").exists()).toBe(false);
-    component.find(".tag-selector__input").simulate("focus");
-    expect(component.find(".tag-selector__dropdown").exists()).toBe(true);
+    expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+    fireEvent.focus(screen.getByRole("textbox", { name: "Tags" }));
+    expect(screen.getByRole("listbox")).toBeInTheDocument();
   });
 
-  it("can select existing tags from dropdown", () => {
-    const component = shallow(
+  it("can select existing tags from dropdown", async () => {
+    render(
       <TagSelector
         label="Tags"
         onTagsUpdate={jest.fn()}
         placeholder="Select or create tags"
         tags={tags}
       />
     );
-    component.find(".tag-selector__input").simulate("focus");
-    component.find('[data-testid="existing-tag"]').at(0).simulate("click");
-    expect(
-      component.find('[data-testid="selected-tag"] span').at(0).text()
-    ).toBe("tag1");
+    fireEvent.focus(screen.getByRole("textbox", { name: "Tags" }));
```

Comment:
I'd suggest to use `useEvent` over `fireEvent`.

`userEvent` provides a more comprehensive simulation of user behavior. When a user interacts with the UI, there are a lot of events that get fired beyond just focusing on the input field.

```suggestion
    await userEvent.click(screen.getByRole("textbox", { name: "Tags" }));
```